### PR TITLE
DAT-17896 Trigger functional and harness tests by request

### PR DESCRIPTION
**Pull request by rberezen**
## Impact

- [ ] Bug fix (non-breaking change which fixes expected existing functionality)
- [ ] Enhancement/New feature (adds functionality without impacting existing logic)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

**Comment by MalloD12** [comment]
Does this mean the test execution flow will now change and if we want to trigger Test-harness and Functional Test execution we need to add their respective labels, right? Does this apply to contributor's PRs as well as internal PRs?

Thanks,
Daniel.

**Comment by rberezen** [comment]
> Does this mean the test execution flow will now change and if we want to trigger Test-harness and Functional Test execution we need to add their respective labels, right? Does this apply to contributor's PRs as well as internal PRs?
> 
> Thanks, Daniel.

yes, sir.
and should be the same procedure for contributor's PRs
but unfortunately I can't test this in branch

File: .github/workflows/run-functional-tests.yml [hunk 1]
```diff
@@ -29,6 +31,12 @@ jobs:
     steps:
       - uses: actions/checkout@v4
 
+      - name: Check runFunctionalTests label
+        if: contains(github.event.pull_request.labels.*.name, 'runFunctionalTests')
+        run: |
+          echo "Add the runFunctionalTests label to the PR to trigger the Functional Tests workflow."
+          exit 1
```
Comment (StevenMassaro) [inline .github/workflows/run-functional-tests.yml:38]: What is this supposed to do?

**Comment by MalloD12** [comment]
I have a few more questions:

- Why are functional tests and test harness triggered without the required tags? Is that what you mentioned it wasn't possible to test in this branch? @rberezen 
- I see most of the checks are being executed twice using `pull_request` and `pull_request_target`. What does it mean each of these? Is this the expected behavior running checks for each of them? @jandroav @sayaliM0412 

Thanks,
Daniel.

File: .github/workflows/run-functional-tests.yml (hunk 1 above)
Comment (rberezen) [reply .github/workflows/run-functional-tests.yml]: ah, good catch
changed to "!contains...'
so if no appropriate label - fail step and job

**Review by StevenMassaro — APPROVED** [review]
Looks fine to me. You could move the "Check runTestHarnessTests label" above the checkout step if you cared to.

**Comment by rberezen** [comment]
> I have a few more questions:
> 
> * Why are functional tests and test harness triggered without the required tags? Is that what you mentioned it wasn't possible to test in this branch? @rberezen
> * I see most of the checks are being executed twice using `pull_request` and `pull_request_target`. What does it mean each of these? Is this the expected behavior running checks for each of them? @jandroav @sayaliM0412
> 
> Thanks, Daniel.

1. yes, can't test in branch
2. looking at it, removed   pull_request: types: [labeled] where it is not necessary...checking

**Comment by filipelautert** [comment]
> Looks fine to me. You could move the "Check runTestHarnessTests label" above the checkout step if you cared to.

@rberezen +1 to @StevenMassaro comments. It will save us some extra ticks.

**Comment by rberezen** [comment]
> I have a few more questions:
> 
> * Why are functional tests and test harness triggered without the required tags? Is that what you mentioned it wasn't possible to test in this branch? @rberezen
> * I see most of the checks are being executed twice using `pull_request` and `pull_request_target`. What does it mean each of these? Is this the expected behavior running checks for each of them? @jandroav @sayaliM0412
> 
> Thanks, Daniel.

@MalloD12 should be better now, no more extra run for unit/integrations tests

**Pull request merged**